google/cloud/firestore_v1/base_query.py

```diff
@@ -315,36 +422,66 @@ def where(self, field_path: str, op_string: str, value) -> "BaseQuery":
             modified with the newly added filter.
 
         Raises:
-            ValueError: If ``field_path`` is invalid.
-            ValueError: If ``value`` is a NaN or :data:`None` and
-                ``op_string`` is not ``==``.
+            ValueError: If
+                * ``field_path`` is invalid.
+                * If ``value`` is a NaN or :data:`None` and ``op_string`` is not ``==``.
+                * FieldFilter was passed without using the filter keyword argument.
+                * `And` or `Or` was passed without using the filter keyword argument .
+                * Both the positional arguments and the keyword argument `filter` were passed.
         """
-        field_path_module.split_field_path(field_path)  # raises
 
-        if value is None:
-            if op_string != _EQ_OP:
-                raise ValueError(_BAD_OP_NAN_NULL)
-            filter_pb = query.StructuredQuery.UnaryFilter(
-                field=query.StructuredQuery.FieldReference(field_path=field_path),
-                op=StructuredQuery.UnaryFilter.Operator.IS_NULL,
+        if isinstance(field_path, FieldFilter):
+            raise ValueError(
+                "FieldFilter object must be passed using keyword argument 'filter'"
             )
-        elif _isnan(value):
-            if op_string != _EQ_OP:
-                raise ValueError(_BAD_OP_NAN_NULL)
-            filter_pb = query.StructuredQuery.UnaryFilter(
-                field=query.StructuredQuery.FieldReference(field_path=field_path),
-                op=StructuredQuery.UnaryFilter.Operator.IS_NAN,
+        if isinstance(field_path, BaseCompositeFilter):
+            raise ValueError(
+                "'Or' and 'And' objects must be passed using keyword argument 'filter'"
             )
-        elif isinstance(value, (transforms.Sentinel, transforms._ValueList)):
-            raise ValueError(_INVALID_WHERE_TRANSFORM)
-        else:
-            filter_pb = query.StructuredQuery.FieldFilter(
-                field=query.StructuredQuery.FieldReference(field_path=field_path),
-                op=_enum_from_op_string(op_string),
-                value=_helpers.encode_value(value),
+
+        field_path_module.split_field_path(field_path)  # raises
```

Comment:
Hmm, this seems to have always been there since this code was added: https://github.com/googleapis/python-firestore/blame/main/google/cloud/firestore_v1/base_query.py#L246

Perhaps a leftover debugging info. I can remove it.